src/canvas/canvas.service.ts

```diff
@@ -123,4 +157,31 @@ export class CanvasService {
       (this.produceEmptyCanvasState(filter.canvasId) as CanvasStateEntity) //If state is empty return default one
     );
   }
+
+  public async giveAccess(command: GiveAccessCommand) {
+    const user = await this.userRepository.findOne({
+      where: { id: command.userId },
+    });
+    const canvas = await this.canvasRepository.findOne({
+      where: { id: command.canvasId },
+    });
+    let canvasAccess = await this.canvasAccessRepository.findOne({
+      where: { user: { id: command.userId }, canvas: { id: command.canvasId } },
+    });
+    if (canvasAccess) {
+      return;
+    }
+    canvasAccess = new CanvasAccessEntity();
+    canvasAccess.isOwner = false;
+    canvasAccess.canvas = canvas;
+    canvasAccess.user = user;
+    await this.canvasAccessRepository.save(canvasAccess);
+  }
+
+  public async cancelAccess(command: CancelAccessCommand) {
+    await this.canvasAccessRepository.delete({
+      user: { id: command.userId },
+      canvas: { id: command.canvasId },
```

Comment:
Tutaj to pewnie poleci http 200 jeżeli będziemy próbowali usunąć dostęp do nieistniejącego canvasu ale w sumie to nie jest jakiś problem